### DISCUSSION TITLE
Resource returns wrong entry in case of multiple models in same table

### DISCUSSION THREAD

**Discussion by juhotaipale** [laravel/nova-issues#4117] [Q&A] [answered]
- Laravel Version: 9.9.0
- Nova Version: 4.2.5
- PHP Version: 8.1.5
- Database Driver & Version: php_pdo_sqlsrv_81_ts_x64.dll
- Operating System and Version: Windows 10 Pro 21H2
- Browser type and version: Google Chrome 100.0.4896.127

### Description:
Nova resource is returning first matching result from database table even if the model is using global scope.

Because of legacy integration I have multiple models assigned to one table which is presenting different categorization for models. I have models such as ``ArticleCategory``, ``ArticleGroup`` and ``SupplierCategory``. The table in Microsoft SQL database contains data as following format.

| typename | groupname | groupdesc         |
|----------|-----------|-------------------|
| ART1     | 10        | Phones            |
| ART1     | 20        | Computers         |
| ART2     | 1010      | Smart phones      |
| ART2     | 1020      | Legacy phones     |
| SUPP1    | 10        | General suppliers |

To properly handle this I have created global scope for each model to filter right entities.

**App/Models/ArticleCategory.php**
```php
class ArticleCategory extends Model
{
    public $incrementing = false;

    protected $connection = 'sqlsrv';
    protected $table = 'usergroup';
    protected $primaryKey = 'groupname';
    protected $keyType = 'string';

    protected static function booted()
    {
        static::addGlobalScope('articleCategory', function (Builder $builder) {
            $builder->where('typename', 'art1');
        });
    }
}
```
**App/Models/SupplierCategory.php**
```php
class SupplierCategory extends Model
{
    public $incrementing = false;

    protected $connection = 'sqlsrv';
    protected $table = 'usergroup';
    protected $primaryKey = 'groupname';
    protected $keyType = 'string';

    protected static function booted()
    {
        static::addGlobalScope('supplierCategory', function (Builder $builder) {
            $builder->where('typename', 'supp1');
        });
    }
}
```

**App/Nova/SupplierCategory.php**
```php
class SupplierCategory extends Resource
{
    public static $model = \App\Models\SupplierCategory::class;

    public function title()
    {
        return $this->groupname.' '.$this->groupdesc;
    }

    public function fields(NovaRequest $request)
    {
        return [
            ID::make(null, 'groupname')->sortable(),
            Text::make('Name', 'groupdesc')->sortable(),
        ];
    }
}
```

Now, with this setup when I visit resource http://127.0.0.1:8000/nova/resources/supplier-categories/10, Nova shows the first matching resource from the database without global scope (image below). The resource list at http://127.0.0.1:8000/nova/resources/supplier-categories is showing everything correctly, and Tinker also returns correct entity when running ``SupplierCategory::find(10)``.

![kuva](https://user-images.githubusercontent.com/13137708/165236408-f362979d-f732-4d5c-aeac-1ff884398bc9.png)

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Setup table and models as described.

**Comment by crynobone** [comment, marked as answer]
Nova disables global scope by default in detail query in order to allow the user to access trashed models, however you can re-enable it by adding the following to each resource or `App\Nova\Resource`:

```php
    /**
     * Build a "detail" query for the given resource.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
     * @param  \Illuminate\Database\Eloquent\Builder  $query
     * @return \Illuminate\Database\Eloquent\Builder
     */
    public static function detailQuery(NovaRequest $request, $query)
    {
        static::newModel()->registerGlobalScopes($query);

        return parent::detailQuery($request, $query);
    }
```
